app/scripts/filters/util.js

```diff
@@ -528,6 +528,31 @@ angular.module('openshiftConsole')
       }
       var uri = new URI(url);
       var protocol = uri.protocol();
-      return uri.is('absolute') && (protocol === 'http' || protocol === 'https');      
+      return uri.is('absolute') && (protocol === 'http' || protocol === 'https');
+    };
+  })
+  // currently used to set an alt text value for an env var that has a
+  // valueFrom attribute instead of a value.
+  .filter('altTextForValueFrom', function() {
+    return function(obj) {
+      if(obj.value) {
+        return;
+      }
+      _.each(obj.valueFrom, function(val, key) {
+        console.log(key);
+        if(key === 'configMapKeyRef') {
+          obj.valueAlt = 'Set to the key ' + key + ' in config map ' + val.name;
```

Comment:
What I mean is that what you have doesn't match what's in the environment directive. See my example above for what I think it should be.